### PR TITLE
Fix #287699: The setting of "View > Mark irregular measures" isn't saved into scores

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**

### CODE REVIEW

File: libmscore/scorefile.cpp [hunk 1]
```diff
@@ -159,6 +159,8 @@ void Score::writeMovement(XmlWriter& xml, bool selectionOnly)
       xml.tag("showUnprintable", _showUnprintable);
       xml.tag("showFrames",      _showFrames);
       xml.tag("showMargins",     _showPageborders);
+      if (!_markIrregularMeasures)
+            xml.tag("markIrregularMeasures", _markIrregularMeasures);
```
Comment (Jojo-Schmitz) [inline libmscore/scorefile.cpp]: It only gets written to the score when not set to the default (i.e. when set to `false`, the default being `true`).
I wonder whether the other 3 settings above should also only get written in that case?

Comment (mattmcclinch) [reply libmscore/scorefile.cpp]: Couldn't you just write
```
xml.tag("markIrregularMeasures", _markIrregularMeasures, true);
```
and achieve the same result? You can leave off the third parameter when the default value is `false` (or any other interpretation of `QVariant()`).

Comment (Jojo-Schmitz) [reply libmscore/scorefile.cpp]: Good idea. Doesn't save much though, only now the comparison is done somewhere else (in `xml.tag()`). And still no good way to determine the default, except to look into the constructor.

Comment (mattmcclinch) [reply libmscore/scorefile.cpp]: Well, the comparison is done in `xml.tag()` anyway, which I guess was my point.

File: libmscore/scorefile.cpp [hunk 2]
```diff
@@ -153,12 +153,12 @@ void Score::writeMovement(XmlWriter& xml, bool selectionOnly)
...
+            style().save(xml, true);      // save only differences to builtin style
 
-      xml.tag("showInvisible",   _showInvisible);
-      xml.tag("showUnprintable", _showUnprintable);
-      xml.tag("showFrames",      _showFrames);
-      xml.tag("showMargins",     _showPageborders);
+      xml.tag("showInvisible", _showInvisible, true);
+      xml.tag("showUnprintable", _showUnprintable, true);
+      xml.tag("showFrames", _showFrames, true);
+      xml.tag("showMargins", _showPageborders, false);
```
Comment (Jojo-Schmitz) [inline libmscore/scorefile.cpp]: Looking at the mtest failures seems to make this change not worth the effort

**Comment by MarcSabatella** [comment]
I oppose the change to the default of this important option.  I would suggest removing the second commit.

**Comment by Jojo-Schmitz** [comment]
I'm not a fan too, and with the change of that request to make it a global setting this commit doesn't cut it anyway

**Pull request merged**